src/pipeline.rs

```diff
@@ -1,12 +1,30 @@
-use crate::Event;
+use crate::{transforms::FunctionTransform, Event};
 use futures01::{
     sync::mpsc::{channel, Receiver, SendError, Sender},
-    AsyncSink, Poll, Sink,
+    Async, AsyncSink, Poll, Sink,
 };
+use std::collections::VecDeque;
 
-#[derive(Debug, Clone)]
+#[derive(Derivative, Clone)]
+#[derivative(Debug)]
 pub struct Pipeline {
     inner: Sender<Event>,
+    // We really just keep this around in case we need to rebuild.
+    #[derivative(Debug = "ignore")]
+    inlines: Vec<Box<dyn FunctionTransform>>,
+    enqueued: VecDeque<Event>,
+}
+
+impl Pipeline {
+    fn try_empty_queue(&mut self) -> Poll<(), <Self as Sink>::SinkError> {
```

Comment:
Would this also make sense as `try_flush`? `try_empty_queue` kinda makes me think it might just drop things rather than forwarding them (though the return signature would make it clearer to me that it is sending).